### ISSUE TITLE
Rule Request: Redundant ternary operator for Bools

### ISSUE THREAD

**Issue by sjavora** [realm/SwiftLint#3207]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Several times now I've seen code like this:
```swift
let a = x ? true : false
```

This is redundant and could just be replaced with
```swift
let a = x
```

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Haven't seen this discussed, but seems like a no-brainer to me.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

OK:
```swift
let a: Bool = x
let b: Bool = !x

var isActive: Bool { somethingElse.isActive }
```
Not OK:
```swift
let a: Bool = x ? true : false
let b: Bool = x ? false : true

var isActive: Bool { somethingElse.isActive ? true : false }
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just severity.

4. Should the rule be opt-in or enabled by default? Why?

I don't believe this should be opt-in.

**Comment by christiankm** [comment]
Making a custom rule should be simple here, at least just for detecting that exact string.
I can try to make a PR for it to be part of the actual library.
